index.bs

```diff
@@ -235,8 +252,26 @@ To <dfn>sanitize an attribute</dfn> named |attr|, run these steps:
 6. if |config| has a non-empty [=attribute allow list=] and |name| is not in |config|'s [=attribute allow list=] return 'drop'
 7. return 'keep'
 
+To <dfn>create a document fragment</dfn>
+named |fragment| from a Sanitizer |input|, run these steps:
+
+1. Switch based on |input|'s type:
+  1. if |input| is of type {{DocumentFragment}}, return |input|.
+  2. if |input| is of type {{Document}}
+    1. let `f` be the result of {{createDocumentFragment}}.
+    2. return the result of `input.cloneNode(input, true)`.
+  3. if |input| is of type `DOMString`, then:
```

Comment:
Why parse into document *and* transform into a document fragment?

Reply:
As you guessed: I tried to find a parse-into-DocumentFragment algorithm in the spec, but didn't find any. Hence, the two-step process. I think our implementation can also do that directly; I just didn't find a clean way to spec it.

(Generally, I'm having a hard time finding good references for spec purposes. I could very well be overlooking something here. My understanding is that, as long as the behaviour is unambiguous we're good to go, and the implementations would certainly be free to pick simpler algorithms.)

Reply:
As I read it, cloneNode is conveniently specified in a way to only copy the structure.